protocols/ledger/ledger_test.go

```diff
@@ -0,0 +1,142 @@
+		}},
+		TurnNum: 0,
+		IsFinal: false,
+	}
+
+	gotState := ledger.SignedStateForTurnNum[0].State()
+	if diff := cmp.Diff(expectedState, gotState); diff != "" {
+		t.Errorf("TestCreateLedger: ledger state mismatch (-want +got):\n%s", diff)
+	}
+
+	ledger2, _ := ledgerManager.CreateTestLedger(left, right, &alice.privateKey, 0)
+	if ledger2.ChannelNonce.Cmp(big.NewInt(1)) != 0 {
+		t.Error("TestCreateLedger: ledger channel should use the next nonce")
+	}
+
+}
+
+func TestHandleLedgerRequest(t *testing.T) {
+	ledgerManager := NewLedgerManager()
+	left := outcome.Allocation{Destination: alice.destination, Amount: big.NewInt(3)}
+	right := outcome.Allocation{Destination: bob.destination, Amount: big.NewInt(2)}
+
+	ledger, _ := ledgerManager.CreateTestLedger(left, right, &alice.privateKey, 0)
+
+	destination := types.AddressToDestination(common.HexToAddress(`0x5e29E5Ab8EF33F050c7cc10B5a0456D975C5F88d`))
+	asset := types.Address{}
+	oId := protocols.ObjectiveId("Test")
+
+	validRequest := protocols.LedgerRequest{ObjectiveId: oId, LedgerId: ledger.Id, Left: left.Destination, Right: right.Destination, Destination: destination, Amount: types.Funds{asset: big.NewInt(4)}}
+	invalidRequest := protocols.LedgerRequest{ObjectiveId: oId, LedgerId: ledger.Id, Left: left.Destination, Right: right.Destination, Destination: destination, Amount: types.Funds{asset: big.NewInt(10)}}
+
+	_, err := ledgerManager.HandleRequest(ledger, validRequest, &alice.privateKey)
+	if err == nil {
+		t.Errorf("TestHandleLedgerRequest: expected request to be fail as there is no supported state")
+	}
+
+	SignPreAndPostFundingStates(ledger, []*[]byte{&alice.privateKey, &bob.privateKey})
+
+	_, err = ledgerManager.HandleRequest(ledger, invalidRequest, &alice.privateKey)
+	if err == nil {
+		t.Errorf("TestHandleLedgerRequest: expected request to  fail as the ledger does not have enough funds")
+	}
+
+	sideEffects, err := ledgerManager.HandleRequest(ledger, validRequest, &alice.privateKey)
+	if err != nil {
+		t.Error(err)
+	}
+	guarantee, _ := outcome.GuaranteeMetadata{
+		Left:  alice.destination,
+		Right: bob.destination,
+	}.Encode()
+
+	expectedState := state.State{
+		ChainId:           big.NewInt(9001),
+		Participants:      []types.Address{alice.address, bob.address},
+		ChannelNonce:      big.NewInt(0),
+		AppDefinition:     types.Address{},
+		ChallengeDuration: big.NewInt(45),
+		AppData:           []byte{},
+		Outcome: outcome.Exit{outcome.SingleAssetExit{
+			Allocations: outcome.Allocations{
+				outcome.Allocation{
+					Destination: alice.destination,
+					Amount:      big.NewInt(1),
+				},
+				outcome.Allocation{
+					Destination: bob.destination,
+					Amount:      big.NewInt(0),
+				},
+				outcome.Allocation{
+					Destination:    destination,
+					Amount:         big.NewInt(4),
+					AllocationType: outcome.GuaranteeAllocationType,
+					Metadata:       guarantee,
+				},
+			}}},
+		TurnNum: 2,
+		IsFinal: false}
+
```

Comment:
Seems to be missing a `gotState, _ := ledger.LatestSupportedState()` + comparison between got and expected.

As far as I can tell, the `NewSignedState` and `SignAndAdd` happen to that state inside of `HandleRequest` and return errors there - the below check may be unnecessary?

Reply:
> Seems to be missing a `gotState, _ := ledger.LatestSupportedState()` + comparison between got and expected.

Good point!

> As far as I can tell, the `NewSignedState` and `SignAndAdd` happen to that state inside of `HandleRequest` and return errors there - the below check may be unnecessary?

While I don't expect this to fail throwing away error responses makes me nervous 😛   My preference would be to always fail a test if a function returns us an error, even if it's a helper function.